From: Robert Marko <robimarko@gmail.com>
Date: Fri, 6 Jan 2023 09:56:54 +0000 (+0100)
Subject: iwinfo: devices: add Qualcomm Atheros IPQ8074 WiSoC
X-Git-Url: http://git.openwrt.org/?p=project%2Fiwinfo.git;a=commitdiff_plain;h=5914d7113ecf77de63eb21fc233684d1a1a52ca5

iwinfo: devices: add Qualcomm Atheros IPQ8074 WiSoC

Add detection of the Qualcomm Atheros IPQ8074 WiSoC using the compatible
string from device tree.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---

--- a/hardware.txt
+++ b/hardware.txt
@@ -167,6 +167,7 @@
 0x168c 0x003c 0x1a56 0x1420    0      0  "Qualcomm Atheros" "QCA9862"
 0x168c 0x003c 0x19b6 0xd03c    0      0  "Mikrotik" "R11e-5HacT"
 0x168c 0x003c 0x168c 0x4019    0      0  "Qualcomm Atheros" "IPQ4019"
+0x168c 0x8074 0x168c 0x8074    0      0  "Qualcomm Atheros" "IPQ8074"
 0x168c 0x003c 0x19b6 0xd075    0      0  "Mikrotik" "R11e-5HacD"
 0x168c 0x0040 0x168c 0x0002    0      0  "Qualcomm Atheros" "QCA9990"
 0x168c 0x0046 0x168c 0xcafe    0      0  "Qualcomm Atheros" "QCA9984"
--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -3381,6 +3381,11 @@ static int nl80211_hardware_id_from_fdt(
 		id->device_id = 0x003c;
 		id->subsystem_vendor_id = 0x168c;
 		id->subsystem_device_id = 0x4019;
+	} else if (!strcmp(compat, "qcom,ipq8074-wifi")) {
+		id->vendor_id = 0x168c;
+		id->device_id = 0x8074;
+		id->subsystem_vendor_id = 0x168c;
+		id->subsystem_device_id = 0x8074;
 	} else if (!strcmp(compat, "mediatek,mt7622-wmac")) {
 		id->vendor_id = 0x14c3;
 		id->device_id = 0x7622;
